transcendence-app/src/setup-app.ts

```diff
@@ -0,0 +1,37 @@
+import { INestApplication, ValidationPipe } from '@nestjs/common';
+import * as session from 'express-session';
+import * as ConnectMemcached from 'connect-memcached';
+import * as passport from 'passport';
+import { ConfigService } from '@nestjs/config';
+import { EnvironmentVariables } from './config/env.validation';
+import { WsSessionAdapter } from './chat/ws-session.adapter';
+
+export const setupApp = (app: INestApplication) => {
```

Comment:
I don't get much how/why can we now afford to not initialize the app from `app.module.ts`

Reply:
We need to use the same session middleware for the HTTP and the WebSocket server (the `app.useWebSocketAdapter(new WsSessionAdapter(app, sessionMiddleware));` line).

It's the same thing as before, but we set up the middlewares in this function and call it from the main instead of setting them up at the module level.